src/preprocessing/passes/bv_ackermann.cpp

```diff
@@ -80,7 +78,7 @@ void addLemmaForPair(TNode args1,
 
 void addChildrenToStack(TNode node, std::stack<TNode>* stack)
 {
-  for (TNode arg: node)
+  for (const TNode& arg: node)
```

Comment:
Not sure why you would use a const TNode&: TNode's do not incur any overhead, but TNode& adds another layer of indirection.

Reply:
`TNode arg` and `const TNode&` should compile to the exact same code:
https://godbolt.org/g/97P4b9

Since this is best practice for Nodes, for consistency I'd rather have that pattern used wherever possible. Maybe we should add that to the guidelines?